### PR TITLE
Adjust CopyRecursiveUnitsTestCase to use recursive_conservative

### PR DESCRIPTION
**Pull request by kersommoura**
Pulp 2.18.1 introduced a new flag `recursive_conservative`. Adjust the
test case CopyRecursiveUnitsTestCase to use this new flag.

`recursive_conservative` is set to False by default.

See: https://pulp.plan.io/issues/4152
See: https://pulp.plan.io/issues/4269
See: https://pulp.plan.io/issues/4375

### CODE REVIEW

**Comment by kersommoura** [comment]
Tested against 2.18.1b1 installation RHEL7.6.

@dralley, and @goosemania, please confirm this is the expected behaviour.

File: pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py [hunk 1]
```diff
@@ -199,9 +201,11 @@ def setUpClass(cls):
     def test_recursive(self):
         """Test recursive copy for a repository with rich/weak dependencies.
 
+        In order to copy all dependencies rich and weak ones, the flag
+        ``recurisve_conservative`` has to be set as True.
```
Comment (goosemania) [inline pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py]: @dralley , @kersommoura , I don't think that's right.
If rich dependencies are resolved only for conservative case, I think it's a bug in solver

Comment (kersommoura) [reply pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py]: @goosemania, thanks.

Comment (dralley) [reply pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py]: @kersommoura Could you add the "True, False" test case here? I plan to look into this bug this afternoon hopefully.

Comment (dralley) [reply pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py]: Here's a diff.  I fixed the issue -- these tests pass with my new PR.

```
--- a/pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py
+++ b/pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py
@@ -201,8 +201,22 @@ class CopyRecursiveUnitsTestCase(unittest.TestCase):
     def test_recursive(self):
         """Test recursive copy for a repository with rich/weak dependencies.
 
-        In order to copy all dependencies rich and weak ones, the flag
-        ``recurisve_conservative`` has to be set as True.
+        See :meth:`do_test`."
+        """
+        repo = self.do_test(True, False)
+        dst_unit_ids = [
+            unit['metadata']['name'] for unit in
+            search_units(self.cfg, repo, {'type_ids': ['rpm']})
+        ]
+        self.assertEqual(
+            len(dst_unit_ids),
+            RPM2_RICH_WEAK_DATA['total_installed_packages'],
+            dst_unit_ids
+        )
+
+    def test_recursive_conservative(self):
+        """Test recursive copy for a repository with rich/weak dependencies.
+
         See :meth:`do_test`."
         """
         repo = self.do_test(True, True)
```

**Comment by kersommoura** [comment]
Tested against ``Platform Version: 2.18.1b1`` RHEL7.6

**Comment by kersommoura** [comment]
Tested against `Platform Version: 2.17.1` RHEL7.6.

**Pull request merged**